### PR TITLE
LibJS: Date additions and fixes

### PR DESCRIPTION
**Pull request by IdanHo**
This fixes 141 test262 test cases :D

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/DatePrototype.cpp [hunk 1]
```diff
@@ -51,17 +52,27 @@ void DatePrototype::initialize(GlobalObject& global_object)
...
     define_native_function(vm.names.setMinutes, set_minutes, 3, attr);
     define_native_function(vm.names.getMonth, get_month, 0, attr);
+    define_native_function(vm.names.setMonth, set_month, 2, attr);
     define_native_function(vm.names.getSeconds, get_seconds, 0, attr);
     define_native_function(vm.names.setSeconds, set_seconds, 2, attr);
     define_native_function(vm.names.getTime, get_time, 0, attr);
+    define_native_function(vm.names.setTime, set_time, 1, attr);
+    define_native_function(vm.names.getTimezoneOffset, get_timezone_offset, 0, attr);
     define_native_function(vm.names.getUTCDate, get_utc_date, 0, attr);
+    define_native_function(vm.names.setUTCDate, set_date, 1, attr); // FIXME: This is a hack, Serenity doesnt currently support timezones other than UTC.
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/DatePrototype.cpp:62]:
```suggestion
    define_native_function(vm.names.setUTCDate, set_date, 1, attr); // FIXME: This is a hack, Serenity doesn't currently support timezones other than UTC.
```

(`LibJS: Add Date.setUTC{Date, Month, Hours, ...}() aliases`)

**Review by linusg — COMMENTED** [review]
Typo in first commit message: `Date::new(GlobalObject&)` -> `now()`

**Comment by IdanHo** [comment]
typos fixed, sorry :sweat_smile:

**Comment by linusg** [comment]
merging of #7846 resulted in conflicts here, sorry about that :sweat_smile:

**Comment by IdanHo** [comment]
> merging of #7846 resulted in conflicts here, sorry about that

rebased :)

**Pull request merged**